rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/41_knn_search_half_byte_quantized.yml

```diff
@@ -731,3 +731,58 @@ setup:
   - match: { hits.hits.0._score: $rescore_score0 }
   - match: { hits.hits.1._score: $rescore_score1 }
   - match: { hits.hits.2._score: $rescore_score2 }
+---
+"Test index configured rescore vector updateable and settable to 0":
+  - requires:
+      cluster_features: ["mapper.dense_vector.rescore_zero_vector"]
+      reason: Needs rescore_zero_vector feature
+
+  - do:
+      indices.create:
+        index: int4_rescore_0_hnsw
+        body:
+          settings:
+            index:
+              number_of_shards: 1
+          mappings:
+            properties:
+              vector:
+                type: dense_vector
+                index_options:
+                  type: int4_hnsw
+                  rescore_vector:
+                    oversample: 0
+
+  - do:
+      indices.create:
+        index: int4_rescore_update_hnsw
+        body:
+          settings:
+            index:
+              number_of_shards: 1
+          mappings:
+            properties:
+              vector:
+                type: dense_vector
+                index_options:
+                  type: int4_hnsw
+                  rescore_vector:
+                    oversample: 1
+
+  - do:
+      indices.put_mapping:
+        index: int4_rescore_update_hnsw
+        body:
+          properties:
+            vector:
+              type: dense_vector
+              index_options:
+                type: int4_hnsw
+                rescore_vector:
+                  oversample: 0
+
+  - do:
+      indices.get_mapping:
+        index: int4_rescore_update_hnsw
+
+  - match: { .int4_rescore_update_hnsw.mappings.properties.vector.index_options.rescore_vector.oversample: 0 }
```

Comment:
We could add some additional conditions for the tests and maybe unify some of them:
- Create an index without rescore_vector
- Do a knn query on it, store results (1)
- Do a knn query with rescore_vector, store results (2)
- Update the index with rescore_vector
- Do a knn query on it without rescore_vector, compare to knn query with rescore_vector results (2) and script_score (2)
- Update the index with rescore_vector 0
- Do a knn query on it without rescore_vector, compare to knn query without rescore_vector results (1)